src/stan/math/prim/mat/functor/finite_diff_gradient.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_GRADIENT_HPP
+#define STAN__MATH__PRIM__MAT__FUNCTOR__FINITE_DIFF_GRADIENT_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+
+namespace stan {
+  
+  namespace math {
+
+    /**
+     * Calculate the value and the gradient of the specified function
+     * at the specified argument using finite difference.  
+     *
+     * <p>The functor must implement 
+     * 
+     * <code>
+     * double
+     * operator()(const
+     * Eigen::Matrix<double,Eigen::Dynamic,1>&)
+     * </code>
+     *
+     * Error should be on order of epsilon ^ 6
+     * 
+     * @tparam F Type of function
+     * @param[in] f Function
+     * @param[in] x Argument to function
+     * @param[out] fx Function applied to argument
+     * @param[out] grad_fx Gradient of function at argument
+     * @param[in] epsilon perturbation size
+     */
+    template <typename F>
+    void
+    finite_diff_gradient(const F& f,
+                         const Eigen::Matrix<double,Eigen::Dynamic,1>& x,
+                         double& fx,
+                         Eigen::Matrix<double,Eigen::Dynamic,1>& grad_fx, 
+                         const double epsilon = 1e-03) {
+      using Eigen::Matrix;
+      using Eigen::Dynamic;
+      Matrix<double,Dynamic,1> x_temp(x);
+
+      int d = x.size();
+      grad_fx.resize(d);
+
+      fx = f(x);
+      
+      for (int i = 0; i < d; ++i){
+        double delta_f = 0.0;
+
+        x_temp(i) += 3.0 * epsilon;
```

Comment:
for clarity, perhaps use:
x_temp(i) = x(i) + 3.0 * epsilon;

this will match the other calls as well.